### PR TITLE
Readme, Contribution, and Pylint

### PR DESCRIPTION
**Pull request by thebadcoder96**
Wowow this bot is building up quickly! :) 

Here are the PR details:

- Updated `readme.md` file to add more information about the bot and its features.
- Added `CONTRIBUTING.md` with guidelines on how to contribute. I have added a section on how to set up the discord running for development. I know @MAdhavbhatia222 has some notes on it and this should be his next PR :)
- Added Linting with `pylint`. Added pylint to our workflow in GitHub Actions as well. Linting will help us write better and more clean code.

@meri-nova I ran pylint and our score was **4.69** but I did some basic best practices like removing trail spaces and breaking down lines of code and our score is now **7.19**. To increase our score, we should add stuff like docstring our functions (add proper comments/documentation). This will also make our code base more readable and set it to a good standard. Anyone who is interested in contributing to start can try to do this and make our code base better :)
>Once this is merged, I can create an issue highlighting this and we can add people to just comment on the issue if they want to work on it.

### CODE REVIEW

File: src/discord_app.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 DISCORD_TOKEN = os.getenv('DISCORD_BOT_TOKEN')
 SETTINGS_CHANNEL_ID = os.getenv('DISCORD_SETTINGS_CHANNEL_ID')
 GENERAL_CHANNEL_ID = os.getenv('DISCORD_GENERAL_CHANNEL_ID')
-DISCORD_SERVER_ID = os.getenv('DISCORD_SERVER_ID')
+DISCORD_SERVER_ID = os.getenv('DISCORD_SERVER_ID') or '0'
```
Comment (KostyaCholak) [inline src/discord_app.py:31]: I guess this line should already be different. @meri-nova didn't we change it to get the values using `os.environ[...]`? I don't think that a default value of zero is a good idea, we should enforce the existence of this env variable by making the program fail with key error, or by explicitly raising an exception explaining that this setting is required.

**Comment by KostyaCholak** [comment]
Thanks for your contribution!!! I'm excited to see people working on this

File: src/discord_app.py (hunk 1 above)
Comment (thebadcoder96) [reply src/discord_app.py:31]: oh totally. I was just messing around trying to get GitHub actions pytest to run. But it seems like the pytest we are using is trying to get these variables from `${{ secrets.DISCORD_SERVER_ID }}`. I think we should rather be mocking these things using pytest mock. I am looking into it.

File: requirements.txt [hunk 2]
```diff
@@ -3,5 +3,8 @@ python-dotenv==1.0.0
 async-lru==2.0.4
 asyncpg==0.29.0
 discord.py==2.3.2
+#below are for testing
```
Comment (KostyaCholak) [inline requirements.txt:6]: Let's put development reqs into a different requirements file (requirements-dev.txt for example).

**Comment by KostyaCholak** [comment]
It would be faster to review and iterate if we split each feature/fix/chore into a separate PR. Then it would be possible to merge what we agree on and discuss the other blocked PRs

File: requirements.txt (hunk 2 above)
Comment (thebadcoder96) [reply requirements.txt:6]: I love the idea of a separate dev and test requirements file.

**Comment by thebadcoder96** [comment]
> It would be faster to review and iterate if we split each feature/fix/chore into a separate PR. Then it would be possible to merge what we agree on and discuss the other blocked PRs

You are right. My intent last night was to only work on the docs but one redbull in I got a little carried away and started doing all this in the same branch. I will stop here and once the god of this repo @meri-nova has blessed us with comments and successful merges, I will create separate PRs for each issue/feature/fix

**Review by meri-nova — CHANGES_REQUESTED** [review]
Hi. Can you delete changes in test_bot.py.

File: src/test_bot.py
[the remark this answers is not in the document]
Comment (thebadcoder96) [reply src/test_bot.py:1]: Roger that

**Review by meri-nova — APPROVED** [review]
Great job!!!!

**Pull request merged**